File: identhub/src/main/java/de/solarisbank/identhub/data/verification/phone/VerificationPhoneDataSourceRepository.java
package de.solarisbank.identhub.data.verification.phone;

import de.solarisbank.identhub.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.data.verification.phone.model.VerificationPhoneResponse;
import de.solarisbank.identhub.domain.verification.phone.VerificationPhoneRepository;
import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;

public class VerificationPhoneDataSourceRepository implements VerificationPhoneRepository {

    private final VerificationPhoneNetworkDataSource verificationPhoneNetworkDataSource;

    public VerificationPhoneDataSourceRepository(VerificationPhoneNetworkDataSource verificationPhoneNetworkDataSource) {
        this.verificationPhoneNetworkDataSource = verificationPhoneNetworkDataSource;
    }

    @Override
    public Single<VerificationPhoneResponse> authorize() {
        return verificationPhoneNetworkDataSource.postAuthorize()
                .subscribeOn(Schedulers.io());
    }

    @Override
    public Single<VerificationPhoneResponse> confirmToken(TransactionAuthenticationNumber token) {
        return verificationPhoneNetworkDataSource.postConfirm(token)
                .subscribeOn(Schedulers.io());
    }
}

File: startup/src/main/java/de/solarisbank/identhub/startup/data/VerificationPhoneDataSourceRepository.kt
package de.solarisbank.identhub.startup.data

import de.solarisbank.identhub.startup.domain.VerificationPhoneRepository
import de.solarisbank.identhub.startup.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.startup.model.VerificationPhoneResponse
import io.reactivex.Single
import io.reactivex.schedulers.Schedulers

class VerificationPhoneDataSourceRepository(
    private val verificationPhoneNetworkDataSource: VerificationPhoneNetworkDataSource
    ) : VerificationPhoneRepository {
    override fun authorize(): Single<VerificationPhoneResponse> {
        return verificationPhoneNetworkDataSource.postAuthorize()
            .subscribeOn(Schedulers.io())
    }

    override fun confirmToken(token: TransactionAuthenticationNumber): Single<VerificationPhoneResponse> {
        return verificationPhoneNetworkDataSource.postConfirm(token)
            .subscribeOn(Schedulers.io())
    }
}